test/namespace.test.js

```diff
@@ -0,0 +1,57 @@
+var API = require('../src/api')
+var assert = require('assert')
+
+describe('Namespace', function() {
+
+  it ('can namespace a route', function() {
+    var api = new API()
+    var foo = api.namespace('foo')
+    var bar = foo.namespace('bar')
+
+    assert.deepEqual(bar.segments, [ 'foo', 'bar' ])
```

Comment:
I might rename `foo` and `bar` to `parent` and `child` or `base` and `nested`. It wasn't clear at first that this nesting namespaces.